sdk/core/azure-core-test/src/main/java/com/azure/core/test/models/RecordingRedactor.java

```diff
@@ -20,9 +20,21 @@
     private static final String REDACTED_UTF_8 = Base64.getEncoder().encodeToString(REDACTED.getBytes(StandardCharsets.UTF_8));
 
     private static final Pattern ACCESS_TOKEN_KEY_PATTERN = Pattern.compile("(?:\"accessToken\":\")(.*?)(?:\",|\"})");
+    private static final Pattern AUTH_HEADER_KEY_PATTERN = Pattern.compile("(?:\"authHeader\":\")(.*?)(?:\",|\"})");
```

Comment:
Given that many of these patterns are simply looking for a JSON property should we combine them into a single pattern?

```java
private static final Pattern JSON_PROPERTY_PATTERN = Pattern.compile("(?:\"(?:<JSPNSP>)\")(.*?)(?:\",|\"})");
```

Where JSPNSP means JSON property name separated by pipe.

Reply:
Another option is moving these into a static constructor and maintaining a list/array of JSON property names to redact and generated the pattern at construction time.